File: factorial/ocaml_factorial/factorial.ml
let rec factorial(n) = if n = 1 then 1 else n * factorial(n - 1);;

let main () =
  if  Array.length Sys.argv <> 1 then
    begin
      Printf.printf "%s <noargs>\n" Sys.argv.(0);
      exit 1
    end
  else
    begin
    for idx_i = 1 to 15 do
      Printf.printf "factorial(%d)=<%d>\n" idx_i (factorial(idx_i))
    done;

    end

let _ = main ()

File: factorial/python_factorial/factorial.py
import sys
#import sys, re, os

def factorial(n):
  if n == 0:
    return 0
  elif n == 1:
    return 1
  else:
    return (n * factorial(n - 1))

def main():
  if len(sys.argv) != 1:
    print ("Usage: %s <noargs>" % sys.argv[0])
    sys.exit()
  for idx_i in range(1, 14):
    print ("factorial(%d)=<%d>" % (idx_i, factorial(idx_i)))
  sys.exit(), end

main()
